bot/cogs/crashes.py

```diff
@@ -1,55 +1,29 @@
 from fred_core_imports import *
 from libraries import common, createembed
 
-import errno
-from functools import wraps
 import nextcord.ext.commands as commands
-from PIL import Image, ImageEnhance
+from PIL import Image, ImageEnhance, UnidentifiedImageError
 from pytesseract import image_to_string, TesseractError
 import zipfile
-from urllib.request import urlopen
 from concurrent.futures import ThreadPoolExecutor
 from time import strptime
+from typing import AsyncIterator
 
 
-
-def timeout(seconds=2, error_message=os.strerror(errno.ETIME)):
-    def decorator(func):
-        def _handle_timeout(signum, frame):
-            raise TimeoutError(error_message)
-
-        def wrapper(*args, **kwargs):
-            signal.signal(signal.SIGALRM, _handle_timeout)
-            signal.setitimer(signal.ITIMER_REAL, seconds)  # used timer instead of alarm
-            try:
-                result = func(*args, **kwargs)
-            finally:
-                signal.alarm(0)
-            return result
-
-        return wraps(func)(wrapper)
-
-    return decorator
-
-
-@timeout(5)
-def timedregexsearch(*args, **kwargs):
+async def regex_with_timeout(*args, **kwargs):
     try:
-        return re.search(*args, **kwargs)
-    except TimeoutError:
-        raise TimeoutError(f"The following regexp timed out: '{args[0]}'")
+        return await asyncio.wait_for(asyncio.to_thread(re.search, *args, **kwargs), 5)
+    except asyncio.TimeoutError:
+        logging.error(f"A regex timed out! ({args[0]}) with flags {kwargs['flags']} on text of length {len(args[1])}")
```

Comment:
Not reraising the error means I won't get an error message in the dev Discord server